migration/migrationmanager: TestPauseMigration failed

migration/migrationmanager.TestPauseMigration [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3168969&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3168969&tab=artifacts#/) on master @ [0a48b0b74b0a6057f1d418875b97830359a52ec6](https://github.com/cockroachdb/cockroach/commits/0a48b0b74b0a6057f1d418875b97830359a52ec6):


```
=== RUN   TestPauseMigration
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestPauseMigration012197834
    test_log_scope.go:74: use -show-logs to present logs inline
    manager_external_test.go:475: error executing 'RESUME JOB $1': pq: job with status pause-requested cannot be resumed
E210710 06:47:58.339345 13772 jobs/registry.go:615  [-] 1  error expiring job sessions: expire-sessions: context canceled
E210710 06:47:58.339478 13772 jobs/registry.go:623  [-] 2  failed to serve pause and cancel requests: context canceled
E210710 06:47:58.339886 13774 jobs/registry.go:635  [-] 3  error claiming jobs: could not query jobs table: claim-jobs: context canceled
E210710 06:47:58.340110 13774 jobs/registry.go:648  [-] 4  error processing claimed jobs: could not query for claimed jobs: select-running/get-claimed-jobs: context canceled
    panic.go:613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestPauseMigration012197834
--- FAIL: TestPauseMigration (0.48s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestPauseMigration PKG=./pkg/migration/migrationmanager TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</p>
</details>
/cc @cockroachdb/kv ajwerner
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestPauseMigration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

Epic: CRDB-7912

migration/migrationmanager.TestPauseMigration [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3449971&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3449971&tab=artifacts#/) on master @ [e4974b1bdab2fb71552b98294e88f43a3d7f551c](https://github.com/cockroachdb/cockroach/commits/e4974b1bdab2fb71552b98294e88f43a3d7f551c):


```
I210915 20:20:44.084882 14376 1@server/server.go:1556  [n2] 75  node connected via gossip
I210915 20:20:44.089559 13111 kv/kvserver/stores.go:269  [n1] 76  wrote 1 node addresses to persistent storage
I210915 20:20:44.171372 12783 1@util/log/event_log.go:32  [n2] 77 ={"Timestamp":1631737244171368812,"EventType":"node_join","NodeID":2,"StartedAt":1631737244040416344,"LastUp":1631737244040416344}
I210915 20:20:44.171496 12783 sql/sqlliveness/slinstance/slinstance.go:293  [n2] 78  starting SQL liveness instance
I210915 20:20:44.171990 14650 sql/temporary_schema.go:500  [n2] 79  skipping temporary object cleanup run as it is not the leaseholder
I210915 20:20:44.172092 14650 sql/temporary_schema.go:501  [n2] 80  completed temporary object cleanup job
I210915 20:20:44.172127 14650 sql/temporary_schema.go:639  [n2] 81  temporary object cleaner next scheduled to run at 2021-09-15 20:50:44.171940665 +0000 UTC
I210915 20:20:44.172144 14655 sql/sqlstats/persistedsqlstats/provider.go:142  [n2] 82  starting sql-stats-worker with initial delay: 52m45.268858861s
I210915 20:20:44.191155 14648 sql/sqlliveness/slstorage/slstorage.go:393  [n2] 83  inserted sqlliveness session 32701732a93c4a019eacd2555bfbeac9
I210915 20:20:44.191436 14648 sql/sqlliveness/slinstance/slinstance.go:180  [n2] 84  created new SQL liveness session 32701732a93c4a019eacd2555bfbeac9
W210915 20:20:44.290125 13178 kv/range_lookup.go:243  [n2,rangefeed=table-stats-cache,range-lookup=/Meta2/Table/20/NULL] 85  range lookup of key /Meta2/Table/20/NULL found only non-matching ranges []; retrying
W210915 20:20:44.292623 13177 kv/range_lookup.go:243  [n2,rangefeed=table-stats-cache,range-lookup=/Table/20] 86  range lookup of key /Table/20 found only non-matching ranges []; retrying
I210915 20:20:44.298155 12783 server/server_sql.go:1093  [n2] 87  done ensuring all necessary startup migrations have run
I210915 20:20:44.301079 14142 jobs/job_scheduler.go:357  [n2] 88  waiting 2m0s before scheduled jobs daemon start
I210915 20:20:44.301655 14144 server/auto_upgrade.go:44  [n2] 89  auto upgrade disabled by testing
I210915 20:20:44.336172 12783 1@server/server.go:2338  [n2] 90  serving sql connections
I210915 20:20:44.381165 14745 util/log/event_log.go:32  [n1,client=127.0.0.1:51932,hostssl,user=root] 91 ={"Timestamp":1631737244356260731,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"kv.range_merge.queue_enabled\" = false","Tag":"SET CLUSTER SETTING","User":"root","SettingName":"kv.range_merge.queue_enabled","Value":"false"}
I210915 20:20:44.411016 14745 migration/migrationmanager/manager.go:115  [n1,client=127.0.0.1:51932,hostssl,user=root,migration-mgr] 92  migrating cluster from 41.0 to 42.0 (stepping through [42.0])
I210915 20:20:44.413560 14745 migration/migrationcluster/cluster.go:118  [n1,client=127.0.0.1:51932,hostssl,user=root,migration-mgr] 93  executing validate-cluster-version=42.0 on nodes n{1,2}
I210915 20:20:44.482843 14745 migration/migrationmanager/manager.go:135  [n1,client=127.0.0.1:51932,hostssl,user=root,migration-mgr] 94  stepping through 42.0
I210915 20:20:44.495967 14745 jobs/registry.go:345  [n1,client=127.0.0.1:51932,hostssl,user=root,migration-mgr] 95  scheduled jobs [693589916001763329]
I210915 20:20:44.497425 15041 jobs/adopt.go:247  [-] 96  job 693589916001763329: resuming execution
I210915 20:20:44.507177 15060 jobs/registry.go:1150  [n1] 97  MIGRATION job 693589916001763329: stepping through state running with error: <nil>
I210915 20:20:44.571709 15008 jobs/jobs.go:509  [n1,client=127.0.0.1:51948,hostssl,user=root] 98  job 693589916001763329: pause requested recorded with reason 
I210915 20:20:44.592877 13553 jobs/adopt.go:469  [-] 99  job 693589916001763329, session e9b887c708f44437848b82a41b399e7f: paused
I210915 20:20:44.609064 15008 migration/migrationmanager/manager.go:115  [n1,client=127.0.0.1:51948,hostssl,user=root,migration-mgr] 100  migrating cluster from 41.0 to 42.0 (stepping through [42.0])
I210915 20:20:44.609642 15008 migration/migrationcluster/cluster.go:118  [n1,client=127.0.0.1:51948,hostssl,user=root,migration-mgr] 101  executing validate-cluster-version=42.0 on nodes n{1,2}
I210915 20:20:44.610773 15008 migration/migrationmanager/manager.go:135  [n1,client=127.0.0.1:51948,hostssl,user=root,migration-mgr] 102  stepping through 42.0
I210915 20:20:44.627017 15008 migration/migrationmanager/manager.go:372  [n1,client=127.0.0.1:51948,hostssl,user=root,migration-mgr] 103  found existing migration job 693589916001763329 for version 42.0 in status pause-requested, waiting
I210915 20:20:44.627315 15008 jobs/registry.go:345  [n1,client=127.0.0.1:51948,hostssl,user=root,migration-mgr] 104  scheduled jobs [693589916001763329]
    manager_external_test.go:476: error executing 'RESUME JOB $1': pq: job with status pause-requested cannot be resumed
I210915 20:20:44.633204 12783 testutils/testcluster/testcluster.go:101  [-] 105  TestCluster quiescing nodes
W210915 20:20:44.633365 14648 sql/sqlliveness/slinstance/slinstance.go:217  [n2] 106  exiting heartbeat loop
W210915 20:20:44.633489 13414 sql/sqlliveness/slinstance/slinstance.go:217  [n1] 107  exiting heartbeat loop
W210915 20:20:44.634213 14762 kv/txn.go:687  [intExec=expire-sessions] 108  failure aborting transaction: node unavailable; try another peer; abort caused by: query execution canceled
W210915 20:20:44.634417 13553 jobs/registry.go:822  [-] 109  canceling all adopted jobs due to stopper quiescing
W210915 20:20:44.634628 14705 kv/txn.go:687  [-] 110  failure aborting transaction: node unavailable; try another peer; abort caused by: expire-sessions: context canceled
E210915 20:20:44.634800 14705 jobs/registry.go:772  [-] 111  error expiring job sessions: expire-sessions: context canceled
W210915 20:20:44.634979 14325 kv/kvserver/intentresolver/intent_resolver.go:790  [-] 112  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
W210915 20:20:44.635797 14745 kv/txn.go:687  [n1,client=127.0.0.1:51932,hostssl,user=root] 113  failure aborting transaction: node unavailable; try another peer; abort caused by: polling for queued jobs to complete: poll-show-jobs: context canceled
W210915 20:20:44.636352 15008 kv/txn.go:687  [n1,client=127.0.0.1:51948,hostssl,user=root,migration-mgr] 114  failure aborting transaction: node unavailable; try another peer; abort caused by: context canceled
W210915 20:20:44.637804 15008 kv/txn.go:687  [n1,client=127.0.0.1:51948,hostssl,user=root] 115  failure aborting transaction: node unavailable; try another peer; abort caused by: job 693589916001763329 could not be loaded: context canceled
W210915 20:20:44.638029 14745 kv/txn.go:687  [n1,client=127.0.0.1:51932,hostssl,user=root] 116  failure aborting transaction: node unavailable; try another peer; abort caused by: connExecutor closing
W210915 20:20:44.637206 14705 kv/txn.go:687  [-] 117  failure aborting transaction: node unavailable; try another peer; abort caused by: context canceled
E210915 20:20:44.638627 14705 jobs/registry.go:780  [-] 118  failed to serve pause and cancel requests: context canceled
W210915 20:20:44.638812 14705 jobs/registry.go:822  [-] 119  canceling all adopted jobs due to stopper quiescing
W210915 20:20:44.641978 15008 kv/txn.go:687  [n1,client=127.0.0.1:51948,hostssl,user=root] 120  failure aborting transaction: node unavailable; try another peer; abort caused by: connExecutor closing
    panic.go:613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestPauseMigration015073309
--- FAIL: TestPauseMigration (1.42s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestPauseMigration PKG=./pkg/migration/migrationmanager TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/kv @cockroachdb/sql-schema ajwerner
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestPauseMigration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/migration/migrationmanager/manager_external_test.go
+++ b/pkg/migration/migrationmanager/manager_external_test.go
@@ -13,6 +13,7 @@ package migrationmanager_test
 import (
 	"context"
 	gosql "database/sql"
+	"fmt"
 	"sort"
 	"sync/atomic"
 	"testing"
@@ -473,6 +474,12 @@ SELECT id
 		t.Fatalf("did not expect the job to run again")
 	case <-time.After(10 * time.Millisecond):
 	}
+	// Wait for the job to actually be paused as opposed to waiting in
+	// pause-requested. There's a separate issue to make PAUSE wait for
+	// the job to be paused, but that's a behavior change is better than nothing.
+	tdb.CheckQueryResultsRetry(t, fmt.Sprintf(
+		`SELECT status FROM crdb_internal.jobs WHERE job_id = %d`, id,
+	), [][]string{{"paused"}})
 	tdb.Exec(t, "RESUME JOB $1", id)
 	ev = <-ch
 	close(ev.unblock)

